File: java/examples/BasicJobSubmission.java
import nordugrid.arc.Logger;
import nordugrid.arc.LogStream_ostream;
import nordugrid.arc.UserConfig;
import nordugrid.arc.Endpoint;
import nordugrid.arc.EndpointList;
import nordugrid.arc.Job;
import nordugrid.arc.JobList;
import nordugrid.arc.JobDescription;
import nordugrid.arc.JobDescriptionList;
import nordugrid.arc.Submitter;
import nordugrid.arc.JobInformationStorageXML;

public class BasicJobSubmission
{
    public static void main(String argv[])
    {
        // Set up logging to stderr with level VERBOSE (a lot of output will be shown)
        LogStream_ostream logstdout = new LogStream_ostream(nordugrid.arc.arc.getStdout());
        logstdout.setFormat(nordugrid.arc.LogFormat.ShortFormat);
        Logger.getRootLogger().addDestination(logstdout);
        Logger.getRootLogger().setThreshold(nordugrid.arc.LogLevel.VERBOSE);
        Logger logger = new Logger(Logger.getRootLogger(), "jobsubmit");

        // UserConfig contains information on credentials and default services to use.
        // This form of the constructor is necessary to initialise the local job list.
        UserConfig usercfg = new UserConfig("", "");
      
        // Simple job description which outputs hostname to stdout
        String jobdesc = "&(executable=/bin/hostname)(stdout=stdout)";
      
        // Parse job description
        JobDescriptionList jobdescs = new JobDescriptionList();
        if (!JobDescription.Parse(jobdesc, jobdescs).toBool()) {
          logger.msg(nordugrid.arc.LogLevel.ERROR, "Invalid job description");
          System.exit(1);
        }
      
        // Use top-level NorduGrid information index to find resources
        Endpoint index = new Endpoint("ldap://index1.nordugrid.org:2135/Mds-Vo-name=NorduGrid,o=grid",
                                      Endpoint.CapabilityEnum.REGISTRY,
                                      "org.nordugrid.ldapegiis");
        EndpointList services = new EndpointList();
        services.add(index);
      
        // Do the submission
        JobList jobs = new JobList();
        Submitter submitter = new Submitter(usercfg);
        if (!submitter.BrokeredSubmit(services, jobdescs, jobs).equals(nordugrid.arc.SubmissionStatus.SubmissionStatusType.NONE)) {
          logger.msg(nordugrid.arc.LogLevel.ERROR, "Failed to submit job");
          System.exit(1);
        }
      
        // Write information on submitted job to local job list (~/.arc/jobs.xml)
        JobInformationStorageXML jobList = new JobInformationStorageXML(usercfg.JobListFile());
        if (!jobList.Write(jobs)) {
          logger.msg(nordugrid.arc.LogLevel.WARNING, "Failed to write to local job list " + usercfg.JobListFile());
        }
      
        // Job submitted ok
        System.out.println("Job submitted with job id " + jobs.begin().next().getJobID());
        return;
    }
}

File: python/examples/basic_job_submission.py
import arc
import sys

# Set up logging to stderr with level VERBOSE (a lot of output will be shown)
logstdout = arc.LogStream(sys.stdout)
logstdout.setFormat(arc.ShortFormat)
arc.Logger_getRootLogger().addDestination(logstdout)
arc.Logger_getRootLogger().setThreshold(arc.VERBOSE)
logger = arc.Logger(arc.Logger_getRootLogger(), "jobsubmit")

# UserConfig contains information on credentials and default services to use.
# This form of the constructor is necessary to initialise the local job list.
usercfg = arc.UserConfig("", "")

# Two simple job descriptions which output hostname to stdout
jobdescstring = "+(&(executable=/bin/hostname)(stdout=stdout))(&(executable=/bin/hostname)(stdout=stdout))"

# Parse job description
jobdescs = arc.JobDescriptionList()
if not arc.JobDescription_Parse(jobdescstring, jobdescs):
    logger.msg(arc.ERROR, "Invalid job description")
    sys.exit(1)

# Use 'arc.JobDescription_ParseFromFile("helloworld.xrsl", jobdescs)'
# to parse job description from file.

# Use top-level NorduGrid information index to find resources
index = arc.Endpoint("nordugrid.org", arc.Endpoint.REGISTRY, "org.nordugrid.archery")
services = arc.EndpointList(1, index)

# Do the submission
jobs = arc.JobList()
submitter = arc.Submitter(usercfg)
if submitter.BrokeredSubmit(services, jobdescs, jobs) != arc.SubmissionStatus.NONE:
    logger.msg(arc.ERROR, "Failed to submit job")
    sys.exit(1)

# Write information on submitted job to local job list (~/.arc/jobs.xml)
jobList = arc.JobInformationStorageSQLite(usercfg.JobListFile())
if not jobList.Write(jobs):
    logger.msg(arc.WARNING, "Failed to write to local job list %s" % usercfg.JobListFile())

# Job submitted ok
sys.stdout.write("Job(s) submitted with job id(s):\n\t%s\n" % '\n\t'.join([str(j.JobID) for j in jobs]))
